frontend/subrequests/types.go

```diff
@@ -0,0 +1,21 @@
+package subrequests
+
+type Request struct {
+	Name        string      `json:"name"`
+	Version     string      `json:"version"`
+	Type        RequestType `json:"type"`
+	Description string      `json:"description"`
+	Opts        []Named     `json:"opts"`
+	Inputs      []Named     `json:"inputs"`
+	Metadata    []Named     `json:"metadata"`
+	Refs        []Named     `json:"refs"`
+}
```

Comment:
I'm wondering about entitlements, but maybe for now it's fine to assume a secure default.

Reply:
entitlements?

Reply:
Maybe I'm wrong, but I was thinking that in the future if you'd want the request to run a container with additional privileges. But like I said, not dealing with it now is still future-proof so it doesn't matter.

Reply:
Then it would be described inside the metadata result.json or a new field. No way to predict what values there would be. I added type field in case some new ideas emerge in the future.